modules/social_features/social_content_block/src/SocialContentBlockOverride.php

```diff
@@ -32,6 +39,10 @@ class SocialContentBlockOverride implements ConfigFactoryOverrideInterface {
    */
   public function __construct(ConfigFactoryInterface $config_factory) {
     $this->configFactory = $config_factory;
+
+    if (\Drupal::hasService('plugin.manager.content_block')) {
```

Comment:
@OH17L Is this condition needed since this service is implemented in the same module?
Also, can we use Dependency Injection here?

Reply:
@ribel it's needed because the config overrides service executing before the process for creating a new service. Also, this is the reason why we can not inject the service